### DISCUSSION TITLE
Suggestions and ideas

### DISCUSSION THREAD

**Discussion by henrichg** [henrichg/PhoneProfilesPlus#62] [Ideas]
Write me your suggestions and ideas, how to improve PPP.

**Comment by andyxo-xo** [comment]
Hi, thanks for your super app!

A suggestion though if I may please... I can't seem to find a way to select & set a beep (notification sound) for when using the widget to switch from an Event to a Manual profile and then back again. Or if just switching between Manual profiles. When it goes from Event to Event then I can select & set a beep happen but not when manually switching. 

So for instance... I am chilling at Home during my Home Event then gotta pop to the store. Ok, I open the widget and select my manual Outdoor profile. The screen brightens, volumes go up etc. But my phone remains quiet, I would like to hear a confirmation beep (of my choosing) that I did in fact switch out of Home Event to Outdoor Manual. And then when I get home, in the widget I press that little arrow symbol top right (BTW, can you please make that arrow symbol BIGGER in the single widget so it's easier to press?) so to get back into Home Event and again would like to hear a confirmation beep (of my choosing).

Hope this is doable in the future and again thanks for your super cool app! Andy

**Reply by henrichg** [reply]
Wait, wait. You want to notify you when you manually activate profile? Yes this not exists. Only end of duration is notified and event start/end. And yes, it is possible to add this.

**Comment by andyxo-xo** [comment]
Yes, this is my suggestion so you can select to have a sound notification
when you manually change profiles, if this is possible please. Thanks :-)

**Reply by henrichg** [reply]
Yes and will be impleneted. But only when user activates profile, not event od end of duration. It is OK for you?

**Comment by andyxo-xo** [comment]
Yes, for when the user activates any manual profile or for when the user
presses the "Restart Event" icon. Thank you :-)

And can you please make the "Restart Event" icon inside the single
Activator widget bigger please? It is easy to press all the profiles inside
the Activator widget but not so easy to press the small "Restart Event"
icon up in the top right corner of the widget...

Again thanks, Andy

**Reply by henrichg** [reply]
OK, also for "Restart events" will be notification.

Icon size: Restart events in Activator and Editor is normal action button in toolbar. It's size is defined by Android itself, not mine.
But is possible to add "Restart events" in Activator into profile list. Then size will be as profile icon size. But, by me, must be added option in Settings/Activator for this: "Display "Restart events" in profile list". It is Ok for you?
But in Editor is better to keep it in toolbar. Because must be possible to restart events also, when displayed are events.

**Comment by andyxo-xo** [comment]
"But, by me, must be added option in Settings/Activator for this: "Display
"Restart events" in profile list". It is Ok for you?"

Yes, this is very good, thank you. I am thinking if "Restart Events" is
then pinned to the top of the single activator widget, with the current
[event] then below that, then followed by the rest of the profiles. So it
would look like this (just my suggestion if you agree):


Restart Events
______________

Current Event [Event]
______________

Profile
Profile
Profile
etc

**Reply by henrichg** [reply]
Currently Activator has top toobar with "Restart events"+"Edit" icons + "trafic light".
In bottom of toolbar is "header" with activated profile+running event
And in bottom of "header" is list of profiles. And in this list will be added, as first item, "Restart events".
Is this ok?

**Comment by andyxo-xo** [comment]
Yes, I think this will be fine if that is the easiest way for you to add
"Restart Events" to that widget list.

I also do not see a way to sort the profiles on the single widget nor on
the Editor Profile page. The order for my profiles is different on the
Editor page than it is in the widget. Perhaps a way to move the profiles
up/down on the Editor page and then the widget will follow this exact order
too (sorry, this is another suggestion).

**Reply by henrichg** [reply]
Widget list? Not Activator? Activator is popup window. Widget is widget in home laucher screen.
Please send screenshot where you want to add "restart events".

Sorting profilles in Editor and widget is also configurable. In Editor switch filter in top bar to "Profiles - show in Activator" and use "handler" (3-line icon at left side of each profile) and drag it up-down.

**Comment by andyxo-xo** [comment]
Hi again, I attached two screenshots:

1) This is my home launcher screen with your green Activator Widget icon
right in the middle (currently set to Home)
2) This is the screen I get when I press that green Activator Widget icon
(I don't know what you call this) but please add the "Restart Events" here
if possible... thank you :-)

"Sorting profiles in Editor and widget is also configurable"
-  Ah yes, I see that now. That is very nice!

**Reply by henrichg** [reply]
And where are screenshots? Are not displayed here. Send me it via e-mail. henrich.gron@gmail.com

**Reply by henrichg** [reply]
In first screenshot is Icon widget. You want to add in it restart events? But is not space for this. This widget displays currently activated proifle. And click to it opens Activator or Editor (it is configurable in PPP Settings/Lauch from widget (Icon, Single row)).

Second screenshot is Activator. In it is possible to add, as first item in profile list, also "Restart events".

**Reply by henrichg** [reply]
If you want "Icon widget" with restart events, use Shortcut. PPP has also support for shortcuts. In shortcut is supported also "restart events".

**Comment by andyxo-xo** [comment]
Hi, I only wish to add a larger "Restart Events" (in profile list as you
say) to the second screenshot (Activator) ... thank you :-)

With the option for sound when pressing "Restart Events" or any profile in
Activator as we discussed earlier.... thanks so much.

**Reply by henrichg** [reply]
OK. ;-)

**Comment by andyxo-xo** [comment]
No, for me one PPP icon widget is enough... no need for two on my
homescreen and prefer to use the activator ;-)

**Comment by henrichg** [comment]
Done. If you want, install this beta version: https://drive.google.com/file/d/1KMasYeFp47knvOJkmQyvuXbcFPNwyF43/view?usp=drivesdk

Parameters:
- Settings/Application interface/Notification of profile activation - all section.
- Settings/Activator/Add "Restart events" into profile list.

**Comment by andyxo-xo** [comment]
Hi, wow, that was fast :-) I will test it out a bit later and send you
feedback... have a great day, Andy

**Comment by y0ungi** [comment]
Hi Henrich, 

first of all, many thanks for this super helpful app! 

During the last years of Covid pandemic and working in home office, one idea for improvement came to my mind, which IMHO would even enhance the app's usefulness: 
When I am at home, but working, I like to manually toggle from "Home"-profile to "Vibrate only". After my working hours, I have to manually switch back to normal mode, but often forget to do so. 
Of course, I could define a time frame with the same settings as my "Home"-profile, where "Vibrate only" is activated, but since I started to work hybrid (home office and on-site) with no fixed on-site day, that does not work. Also, when I have holidays at home it would switch to "Vibrate only"-mode, although I want it to stay in standard "Home"-profile then. 

Therefore, I was thinking if it is possible to add a timer switch? That means, if you long-press on a profile in the Starter, for example, a menu could pop up which allows to set a time period. PPP would activate the chosen profile for the set time period and restarts events at the end.

**Comment by henrichg** [comment]
Use profile activation duration. Profile parameter section "End of
activation". In it you can configure duration or exact time.

**Reply by y0ungi** [reply]
Oh, I did not know about this option. 
I've just tried it and that is exactly what I was looking for. 
Thanks!

**Comment by klopsos** [comment]
Hi!
Thank You for such a great app! I find it very usefull.
Have a few ideas
Could You add an option for a more condensed widget? Presently i use List widget resized to 4x1. It shows me 4 profile icons +restart event icon in a row. Would be great to show more profiles maybe without text, only icons. So it would show 6-7 of them for example.

Also noticed that Notification sensor wont show me system apps under 'Applications' selection.

**Comment by henrichg** [comment]
List widget: In it is one item with min. width = 70dp. I will testing it with smaller width. But with profile name it is problem. Because will not be space for display it. I must test it.

Notification sensor: Yes system applications are not supported. Which system app you want to configure?

**Reply by klopsos** [reply]
Notification sensor is not a problem for me, just something i noticed. Whas trying to check for android auto running, but i can get it with bluetooth checking also, so its fine.
About widget-just hoping to see some icons in a row, text not important.

**Comment by henrichg** [comment]
Better?
[Compat List widget](https://drive.google.com/file/d/1E_2Vzk-o3c_Gjy7P0OTwRlLDPi6GIT_2/view?usp=drivesdk)

**Comment by henrichg** [comment]
Implemented. If you want, install this beta version: 
https://drive.google.com/file/d/1CR4y2u3V8PCwQK9sw1P88JrlpSdA2Ou4/view?usp=drivesdk

In Settings in "Widget: List" is added parameter "Compact grid".

**Reply by klopsos** [reply]
That looks good, thank You very much!
I sent some money you way for a trouble.

**Comment by voxvargr** [comment]
Idk if it's possible but detecting if media/music is playing would be really helpful for me. That's the last straw holding me on tasker or automate instead of switching to foss which I'd like to.

My main reason for this is that I like having my notification and ringer volume lowered automatically when I start watching a YouTube video or listening to music in the car and then restore the volume after media/music stops playing.

**Reply by henrichg** [reply]
Partially implented, but problem is detecting which application plays. It is possible to detect this, but PPP must be as system app. :-( Then sorry, PPP is not system app.

Then: dection of media play in device is implemented, detection, which applicaiton plays, not.

**Reply by voxvargr** [reply]
That's totally fine, if it can detect anything playing but not specific apps that's perfect!

**Reply by henrichg** [reply]
OK, implemented. Please test it. In Events is new sensor "Music sensor".

https://drive.google.com/file/d/1NhPVUdioqLfd11-fQaa9d5ATWy4ubf7P/view?usp=drivesdk

**Reply by voxvargr** [reply]
Just requested access and will test immediately. I'm excited

**Reply by voxvargr** [reply]
Just switched my media playing profile from specific apps opened to the music sensor and it works so far in several media apps including YouTube and 3rd party not well known apps. Will report back if I have any issues but if I'm silent its working flawlessly. Ping if I'm silent for too long and I'll let you know it's working perfectly. Dude you're awesome

**Reply by voxvargr** [reply]
Still loving this, absolutely no issues and works perfectly!

**Reply by henrichg** [reply]
Good, will be in next release.

**Comment by orlproi** [comment]
Hello, I'm not very familiar with forums, so I posted a suggestion in the "Issues" section instead of here. I wanted to know if it was possible to include a condition outside of the selected location in the "Location" section. That is, activate the condition only if you're outside the selected location. I think it's a useful condition.
